templates/indel_classifier.py

```diff
@@ -0,0 +1,268 @@
+        "clonality": clonality
+    }
+
+
+
+def parse_indels(csv_path):
+    try:
+        df = pd.read_csv(csv_path)
+    except Exception as e:
+        print(f"Error reading the CSV file: {e}")
+        sys.exit(1)
+
+    # Ensure string type for columns that will use string methods
+    for column in ['pre_ins_nt', 'ins_nt', 'post_ins_nt']:
+        df[column] = df[column].astype(str)
+
+    # Processing insertions: filter out 'N' and check if DataFrame is empty
+    ins_df = df[df['Modification'] == 'ins']
+    ins_df = ins_df[
+        ~(ins_df['pre_ins_nt'].str.contains('N') |
+          ins_df['ins_nt'].str.contains('N') |
+          ins_df['post_ins_nt'].str.contains('N'))
+    ]
+
+    if ins_df.empty:
+        grouped_ins = pd.DataFrame(columns=['Start', 'Length', 'pre_ins_nt', 'ins_nt', 'post_ins_nt', 'Count'])
+    else:
+        grouped_ins = ins_df.groupby(['Start', 'Length', 'pre_ins_nt', 'ins_nt', 'post_ins_nt']).size().reset_index(name='Count')
+
+    # Process deletions: Filter by 'del'/'delin' and handle empty DataFrame
+    dels_df = df[df['Modification'].isin(['del', 'delin'])]
+    if dels_df.empty:
+        grouped_dels = pd.DataFrame(columns=['Start', 'Length', 'Count'])
+    else:
+        grouped_dels = dels_df.groupby(['Start', 'Length']).size().reset_index(name='Count')
+
+    return grouped_dels, grouped_ins
+
+
+def additional_indels_cols(df):
+    # Calculate percentages for in-frame and out-of-frame deletions and insertions
+    # Initialize the columns to store the sums of outframe and inframe deletions and insertions
+    df['Outframe'] = 0
+    df['Inframe'] = 0
+
+    # Check if the necessary base columns exist before attempting calculations
+    required_columns = ['Dels_inframe', 'Dels_outframe', 'Ins_inframe', 'Ins_outframe', 'Total Reads']
+    if all(col in df.columns for col in required_columns):
+        # Aggregate inframe and outframe mutations
+        df['Inframe'] = df['Dels_inframe'] + df['Ins_inframe']
+        df['Outframe'] = df['Dels_outframe'] + df['Ins_outframe']
+
+        # Calculate the percentage for Inframe and Outframe
+        df['% Inframe'] = (df['Inframe'] / df['Total Reads']).fillna(0) * 100
+        df['% Outframe'] = (df['Outframe'] / df['Total Reads']).fillna(0) * 100
+
+        # Handle any potential division by zero issues by replacing infinities with zero
+        df['% Inframe'] = df['% Inframe'].replace([np.inf, -np.inf], 0)
+        df['% Outframe'] = df['% Outframe'].replace([np.inf, -np.inf], 0)
+    else:
+        # If any essential columns are missing, set default percentage values to zero
+        df['% Inframe'] = 0
+        df['% Outframe'] = 0
+
+    # Now, df contains two new columns: '% Inframe' and '% Outframe' with the calculated percentages.
+    return df
+
+
+
+def main():
+
+    min_read_threshold = 200
+
+    if len(sys.argv) < 3:
+        print("Usage: python script.py indels.csv edits.csv")
+        sys.exit(1)
```

Comment:
When using a Nextflow template we don't need to pass the files as arguments, we can use the variable form the input of the module and they will be interpolated directly into the script
```suggestion
```